### PR TITLE
Implement animals-sales REST API

### PR DESCRIPTION
**Pull request by isubo**

### CODE REVIEW

File: app/service.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        if not center:
+            raise AnimalsSalesServiceException(f'Center with id={id} was not found')
+
+        animals_response = []
+        for animal in self._animals_dao.get_all_by_center_id(id_int):
+            animals_response.append(dict(id=animal['id'], name=animal['name'], species=animal['species']))
+        return dict(id=id_int, name=center['login'], animals=animals_response)
+
+
+class Validator:
```
Comment (ehborisov) [inline app/service.py]: Your static validators will probably suffice for the simple service like yours, however as the validation logic will grow, there will be more code duplication.

I did a similar task some time ago, check how you can use descriptors (`__get__` and `__set__` methods) to extract some common validation logic into some parent validator class and then create validators with various conditions.

https://github.com/ehborisov/otus-homework/blob/master/hw3/scoring_server/api.py#L80

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[tox]
```
[the remark this answers is not in the document]
Comment (isubo) [reply tox.ini]: Fixed

File: tests/test_service.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import unittest
+from unittest.mock import MagicMock
+from service import AnimalsService, AnimalsSalesServiceException
+
+
+class TestAnimalsService(unittest.TestCase):
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (isubo) [reply tests/test_service.py]: Fixed

File: app/service.py (hunk 1 above)
Comment (isubo) [reply app/service.py]: Fixed

File: app/request_inputs.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        if missing_required_fields:
+            raise InputParamsValidationException('Request does not have required fields: %s' %
+                                                 ', '.join(sorted(missing_required_fields)))
+        for param, arg in kwargs.items():
+            if param in self._known_fields:
+                setattr(self, param, arg)
+            else:
+                app_log.info('Got unknown param in request: %s=%s', param, arg)
+
+    def to_dict(self, not_null=False, key_to_lowercase=True):
```
Comment (ehborisov) [inline app/request_inputs.py:147]: that method looks waay to arcane for what it does. It's ok to have it, but in real life having so many clever one-liners is not really beneficial in the long run.

Also, `not not_null` sounds more like `is_null` to me

**Pull request merged**